beacon-chain/rpc/validator_server.go

```diff
@@ -252,3 +252,25 @@ func (vs *ValidatorServer) retrieveActiveValidator(beaconState *pbp2p.BeaconStat
 	}
 	return beaconState.ValidatorRegistry[validatorIdx], nil
 }
+
+// filterActivePublicKeys takes a list of validator public keys and returns
+// the list of active public keys.
+func (vs *ValidatorServer) filterActivePublicKeys(beaconState *pbp2p.BeaconState, pubkeys [][]byte) ([][]byte, error) {
```

Comment:
I dont underatand this method, so why do we retrieve a map from the db and then get the active validators ? Why not just get the active validators from the state and then filter out which are not in the desired pubkey set. This is a very inefficient method as we generate the validator map from scratch whenever we call this method.